docs/reference/ml/trained-models/apis/get-trained-models-stats.asciidoc

```diff
@@ -116,9 +116,11 @@ The desired number of nodes for model allocation.
 (string)
 include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=model-id]
 
-`model_size`:::
+`model_size_bytes`:::
 (<<byte-units,byte value>>)
-The size of the loaded model in bytes.
+The native memory size required to load this model in bytes.
```

Comment:
@lcawl Any thoughts on the phrasing here would be appreciated.

Reply:
> The native memory size required to load this model in bytes.

This isn't true. We set this also for models loaded in Java. And we should not reference disk size unless specifically referencing PyTorch/native. models

Reply:
This is the `model_size_stats` under `deployment_stats`. We don't have that for java models.

Reply:
It's confusing to have 2 fields called `model_size_bytes` meaning different things. Here we want to express how much free space is required to launch the model.

The motivation is that a user may look at the available memory on the node as reported in the UI and question why a 200MB model cannot be deployed on a node with 300MB of free memory. The answer is that more physical memory is required for loading the model.

Reply:
Actually, thinking about this, the problem is the name `model_size_bytes`. This should really be something like `estimated_memory_bytes`. @davidkyle Do you think it's worth changing this for 8.0? It'll help us get rid of a bad name going forward.

Reply:
I don't think we can get away from admitting there are two types of model.  So something like:

```suggestion
If the model is a PyTorch model then this number represents the native memory
size required to load this model in bytes. If the model is a Java model then this
number represents the amount of JVM heap space that the model will consume
in bytes.
```

Maybe not that exact text but it's clear that we're going to have to explain that the meaning varies depending on the type of model.

Reply:
We discussed this offline and decided we should refactor the way we report this. I'll close this PR and open a different one with the new format.